core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -1001,14 +1003,18 @@ public void onFailure(Throwable t)
         }, executor);
     }
 
+    private synchronized void fatalUnacknowledgedFailure(Throwable cause) {
```

Comment:
I think making this a top level method is more likely to cause confusion, because this should _only_ be valid to call from within the failure hander in `doScheduleAsyncCleanupRequest`. Maybe it should be a method defined there and more specifically named something like `failedToCleanUpRemoteTask(TrinoTransportException)` ?

That way, we know that the update is always guarded by the check for `getTaskInfo().getTaskStatus().getState().isDone()` which will avoids the potential for replacing some other final state that `TaskInfo` might have had set on it already and also avoid accumulating multiple redundant `TrinoTransportException` failures being reported.